test/e2e/healthcheck_hybrid_test.go

```diff
@@ -0,0 +1,155 @@
+		Expect(err).ToNot(HaveOccurred())
+
+		universalClusters, err := NewUniversalClusters([]string{Kuma3}, Silent)
+		Expect(err).ToNot(HaveOccurred())
+
+		global = k8sClusters.GetCluster(Kuma1)
+		err = NewClusterSetup().
+			Install(Kuma(core.Global)).
+			Install(YamlK8s(meshMTLSOn("default"))).
+			Setup(global)
+		Expect(err).ToNot(HaveOccurred())
+
+		optsRemoteK8s = []DeployOptionsFunc{
+			WithIngress(),
+			WithGlobalAddress(global.GetKuma().GetKDSServerAddress()),
+		}
+		remoteK8s = k8sClusters.GetCluster(Kuma2)
+		err = NewClusterSetup().
+			Install(Kuma(core.Remote, optsRemoteK8s...)).
+			Install(KumaDNS()).
+			Install(YamlK8s(namespaceWithSidecarInjection(TestNamespace))).
+			Install(DemoClientK8s("default")).
+			Setup(remoteK8s)
+		Expect(err).ToNot(HaveOccurred())
+
+		echoServerToken, err := global.GetKuma().GenerateDpToken("default", "echo-server_kuma-test_svc_8080")
+		Expect(err).ToNot(HaveOccurred())
+		ingressToken, err := global.GetKuma().GenerateDpToken("default", "ingress")
+		Expect(err).ToNot(HaveOccurred())
+
+		optsRemoteUniversal = []DeployOptionsFunc{
+			WithGlobalAddress(global.GetKuma().GetKDSServerAddress()),
+		}
+		remoteUniversal = universalClusters.GetCluster(Kuma3)
+		err = NewClusterSetup().
+			Install(Kuma(core.Remote, optsRemoteUniversal...)).
+			Install(EchoServerUniversal("dp-echo-1", "default", "echo-universal-1", echoServerToken, WithProtocol("tcp"))).
+			Install(EchoServerUniversal("dp-echo-2", "default", "echo-universal-2", echoServerToken, WithProtocol("tcp"), ProxyOnly(), ServiceProbe())).
+			Install(EchoServerUniversal("dp-echo-3", "default", "echo-universal-3", echoServerToken, WithProtocol("tcp"))).
+			Install(IngressUniversal("default", ingressToken)).
+			Setup(remoteUniversal)
+		Expect(err).ToNot(HaveOccurred())
+		err = remoteUniversal.VerifyKuma()
+		Expect(err).ToNot(HaveOccurred())
+	})
+
+	AfterEach(func() {
+		if ShouldSkipCleanup() {
+			return
+		}
+
+		_ = k8s.KubectlDeleteFromStringE(remoteK8s.GetTesting(), remoteK8s.GetKubectlOptions(), namespaceWithSidecarInjection(TestNamespace))
+		err := remoteK8s.DeleteKuma(optsRemoteK8s...)
+		Expect(err).ToNot(HaveOccurred())
+
+		err = remoteUniversal.DeleteKuma(optsRemoteUniversal...)
+		Expect(err).ToNot(HaveOccurred())
+
+		err = global.DeleteKuma()
+		Expect(err).ToNot(HaveOccurred())
+
+		err = remoteUniversal.DismissCluster()
+		Expect(err).ToNot(HaveOccurred())
+	})
+
+	It("should not load balance requests to unhealthy instance", func() {
+		pods, err := k8s.ListPodsE(remoteK8s.GetTesting(), remoteK8s.GetKubectlOptions(TestNamespace), metav1.ListOptions{
+			LabelSelector: fmt.Sprintf("app=%s", "demo-client"),
+		})
+		Expect(err).ToNot(HaveOccurred())
+		Expect(pods).To(HaveLen(1))
+
+		var counter1, counter2, counter3 int
+		for i := 0; i < 100; i++ {
+			var stdout string
+			cmd := []string{"curl", "-v", "-m", "3", "--fail", "echo-server_kuma-test_svc_8080.mesh"}
+			_, err := retry.DoWithRetryE(
```

Comment:
I think this retry is a little bit of cheating. What if somehow we load balance traffic to echo-2? As far as I understand  `command terminated with exit code 56` was supposed to catch that but there can be any other error technically.

Reply:
First attempts have `exit code 28` because of the timeout, but you are right: `command terminated with exit code 56` is supposed to catch load balancing to echo-2, but as you can see test will immediately fail if we receive this error at least once:
```go
if err != nil {
    Expect(err).ToNot(MatchError("command terminated with exit code 56"))
    return "", nil
}
```